src/components/MDXComponents/EmailSubscriptionForm.tsx

```diff
@@ -0,0 +1,144 @@
+
+  const [formData, setFormData] = useState<FormType>({
+    email: "",
+    loading: false,
+    error: null,
+  });
+  const [success, setSuccess] = useState(false);
+
+  const validateEmail = (): boolean => {
+    // Set up a schema for validating the email address
+    const schema = z.string().email();
+
+    try {
+      schema.parse(formData.email);
+      return true;
+    } catch (e) {
+      console.error(e);
+      setFormData({ ...formData, error: "invalidEmail" });
+      return false;
+    }
+  };
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+    setFormData((prevState) => ({ ...prevState, email: value }));
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (validateEmail()) {
+      // Submitting...
+      setFormData({ ...formData, loading: true, error: null });
+
+      try {
+import axios from "axios";
+
+        await axios.post(API_URL, undefined, {
+          email: formData.email
+        });
+
+        setFormData({ ...formData, loading: false, error: null });
+        setSuccess(true);
+      } catch (e) {
+        setFormData({ ...formData, loading: false, error: "networkError" });
+        console.error(e);
+      }
+    }
+  };
+
+  return (
+    <div>
+      <p>
+        <strong>
+          <Trans i18nKey="releaseSubscription.title" />:
+        </strong>
+      </p>
+
+      {!success ? (
+        <div>
+          <form onSubmit={handleSubmit}>
+            <Box display="flex" alignItems="center">
+              <TextField
+                size="small"
+                variant="outlined"
+                type="text"
+                name="email"
+                placeholder={t("releaseSubscription.placeholder")}
+                value={formData.email}
+                onChange={handleChange}
+                error={!!formData.error}
+                helperText={formData.error === "invalidEmail" ? t('releaseSubscription.error.invalidEmail') : t('releaseSubscription.error.networkError')}
+              />
+import { LoadingButton } from "@mui/lab"
+
+              <LoadingButton
+                type="submit"
+                variant="contained"
+                color="primary"
+                loading={formData.loading}
+                loadingIndicator={t('releaseSubscription.button.subscribing')}
+                ml="1rem"
+              >
+                <Trans i18nKey="releaseSubscription.button.subscribe" />
+              </LoadingButton>
+            </Box>
+          </form>
+          {formData.error && (
+            <p sx={{ color: "#ef5350"}}>
+              {formData.error === "invalidEmail" ? (
+                <Trans i18nKey="releaseSubscription.error.invalidEmail" />
+              ) : (
+                <Trans i18nKey="releaseSubscription.error.networkError" />
+              )}
+            </p>
+          )}
```

Comment:
If we use the `error` prop and the `helperText` prop of `TextField`, we can remove this block.